client/src/extension.ts

```diff
@@ -194,26 +198,37 @@ export function activate(context: ExtensionContext) {
     customCommands.switchImplIntf(client);
   });
 
-  // This sets up a listener that, if we're in dead code analysis mode, triggers
-  // dead code analysis as the LS server reports that ReScript compilation has
-  // finished. This is needed because dead code analysis must wait until
-  // compilation has finished, and the most reliable source for that is the LS
-  // server, that already keeps track of when the compiler finishes in order to
-  // other provide fresh diagnostics.
-  client.onReady().then(() => {
-    context.subscriptions.push(
-      client.onNotification("rescript/compilationFinished", () => {
-        if (inDeadCodeAnalysisState.active === true) {
-          customCommands.deadCodeAnalysisWithReanalyze(
-            inDeadCodeAnalysisState.activatedFromDirectory,
-            diagnosticsCollection,
-            diagnosticsResultCodeActions
-          );
-        }
-      })
-    );
+  commands.registerCommand("rescript-vscode.restart_language_server", () => {
+    client.stop().then(() => {
+      client = createLanguageClient();
+      attachDeadCodeAnalysis(client);
+      context.subscriptions.push(client.start());
+    });
   });
 
+  function attachDeadCodeAnalysis(client: LanguageClient) {

```

Comment:
How about moving this function up so it's not in the middle of the logic.

Reply:
@cristianoc moved the function to the top.